src/conditionals/programs_with_decision_points/medium/q6/MainTest.java

```diff
@@ -0,0 +1,88 @@
+package conditionals.programs_with_decision_points.medium.q6;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+  public Clause[] testSentence() {
+    TestOption.isInputTest = true;
+    TestOption.defaultInput = "w";
+    return new Clause[] {
+        new StringLiteral("Enter w, a, s, or d :"),
+        new NewLine(),
+        new PlaceHolder(),
+    };
+  }
+
+  public void runMain() {
+    Gamer.main(new String[0]);
+  }
+
+  static Stream<Arguments> inputProviderInvalidInput() {
+    return Stream.of(
+      Arguments.of("b", "Invalid Input!"),
+      Arguments.of("c", "Invalid Input!"),
+      Arguments.of("e", "Invalid Input!"),
+      Arguments.of("f", "Invalid Input!"),
+      Arguments.of("g", "Invalid Input!"),
+      Arguments.of("h", "Invalid Input!"),
+      Arguments.of("i", "Invalid Input!"),
+      Arguments.of("j", "Invalid Input!"),
+      Arguments.of("k", "Invalid Input!"),
+      Arguments.of("l", "Invalid Input!"),
+      Arguments.of("m", "Invalid Input!"),
+      Arguments.of("n", "Invalid Input!"),
```

Comment:
Just a note for the future, we probably don't need all these test cases, especially not for a Parsons question.